src/replication.c

```diff
@@ -831,34 +1127,93 @@ ask_replica_status_all(spec_t *spec)
 	MTX_UNLOCK(&spec->rq_mtx);
 }
 
-/*
- * This function processes the status response received from replica
- * and update spec's healthy/degrade count
- */
+static void
+handle_prepare_for_rebuild_resp(spec_t *spec, zvol_io_hdr_t *hdr,
+    mgmt_ack_t *ack_data, mgmt_cmd_t *mgmt_cmd)
+{
+
+	int ret = 0;
+	rcommon_mgmt_cmd_t *rcomm_mgmt = mgmt_cmd->rcomm_mgmt;
+	mgmt_ack_t *buf = (mgmt_ack_t *)rcomm_mgmt->buf;
+	
+	if (hdr->status != ZVOL_OP_STATUS_OK) {
+		rcomm_mgmt->cmds_failed++;
+	} else {
+		memcpy(&buf[rcomm_mgmt->cmds_succeeded++], ack_data,
+		    sizeof (mgmt_ack_t));
+	}
+
+	if (rcomm_mgmt->cmds_sent == rcomm_mgmt->cmds_succeeded) {
+		ret = start_rebuild(buf, spec->target_replica, rcomm_mgmt->buf_size);
+		if (ret == 0) {
+			rcomm_mgmt->buf = NULL;
+			REPLICA_LOG("Rebuild triggered on Replica ip:%s port:%d"
+			    " state:%d\n", spec->target_replica->ip,
+			    spec->target_replica->port, spec->target_replica->state);
+		} else {
+			MTX_LOCK(&spec->rq_mtx);
+			spec->target_replica = NULL;
+			spec->rebuild_in_progress = false;
+			MTX_UNLOCK(&spec->rq_mtx);
+			REPLICA_LOG("Unable to trigger rebuild on Replica ip:"
+			    "%s port:%d state:%d\n", spec->target_replica->ip,
+			    spec->target_replica->port, spec->target_replica->state);
+		}	
+		free_rcommon_mgmt_cmd(rcomm_mgmt);
+	} else if (rcomm_mgmt->cmds_sent ==
+	    (rcomm_mgmt->cmds_failed + rcomm_mgmt->cmds_succeeded)) {
+		free_rcommon_mgmt_cmd(rcomm_mgmt);
```

Comment:
some print over here, and, above 'NOT_OK' case

Reply:
Done!